### PR TITLE
feat(YouTube): add `Hide Suggestions shelf` patch

### PR DESCRIPTION
**Pull request by johnconner122**
Continuation of: https://github.com/ReVanced/revanced-patches/pull/2786

closes: ReVanced/revanced-patches-template#494, ReVanced/revanced-patches-template#469

depends on:

 - https://github.com/ReVanced/revanced-integrations/pull/459

**TODO:**
- [x] If we go back to library tab from home tab by pressing back button, navbar index will be still zero and playlist shelf will be hidden. And vice versa, if go back to home tab from library tab, index will be 4, and suggestions shelves will be shown on home page.
- [x] If we switch to incognito mode when we are in library tab, suggestions shelf is shown on home feed.
- [ ] Whenever we are in library tab, and open download page, or history page, or any other page in library tab, and then go back, back pressed will be called and index will go back to previous one, but we are still in library tab, and index shouldn't have changed.

### CODE REVIEW

**Comment by 0xrxL** [comment]
Have you tried to replicate the same issue with uTube? Maybe you done some wrong things on integrations side.

**Comment by johnconner122** [comment]
> Have you tried to replicate the same issue with uTube? 

Playlist shelf gets hidden but comes back after refreshing library feed.

**Comment by 0xrxL** [comment]
> > Have you tried to replicate the same issue with uTube?
> 
> Playlist shelf gets hidden but comes back after refreshing library feed.

Use this as solution so...

**Comment by johnconner122** [comment]
> > > Have you tried to replicate the same issue with uTube?
> > 
> > 
> > Playlist shelf gets hidden but comes back after refreshing library feed.
> 
> Use this as solution so...

Playlist shelf is still hidden even after refreshing by using this PR.

**Comment by 0xrxL** [comment]
> > > > Have you tried to replicate the same issue with uTube?
> > > 
> > > 
> > > Playlist shelf gets hidden but comes back after refreshing library feed.
> > 
> > 
> > Use this as solution so...
> 
> Playlist shelf is still hidden even after refreshing by using this PR.

I meant porting uPatcher solution to this PR.

**Comment by johnconner122** [comment]
> I meant porting uPatcher solution to this PR.

Done, working fine. But now there is another problem, when we switch to incognito mode when we are in library tab, suggestions shelf is shown on home feed.

File: src/main/kotlin/app/revanced/patches/youtube/layout/hide/suggestionsshelf/resource/patch/BreakingNewsResourcePatch.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import app.revanced.patcher.data.ResourceContext
+import app.revanced.patcher.patch.ResourcePatch
+import app.revanced.patcher.patch.annotations.DependsOn
+import app.revanced.patches.shared.mapping.misc.patch.ResourceMappingPatch
+import app.revanced.patches.youtube.layout.hide.suggestionsshelf.patch.SuggestionsShelfPatch
+
+@DependsOn([ResourceMappingPatch::class])
+class BreakingNewsResourcePatch: ResourcePatch {
+    override fun execute(context: ResourceContext) {
+        SuggestionsShelfPatch.horizontalCardListId = ResourceMappingPatch.resourceMappings.single {
```
[the remark this answers is not in the document]
Comment (johnconner122) [reply src/main/kotlin/app/revanced/patches/youtube/layout/hide/suggestionsshelf/resource/patch/BreakingNewsResourcePatch.kt:12]: oops, copy paste

**Comment by johnconner122** [comment]
@LisoUseInAIKyrios, @oSumAtrIX inotia00 found a Bytecode method to hide suggestions shelf:
https://github.com/inotia00/revanced-patches/blob/fc41dd043da193fcd136c0c05534b6db143e2e32/src/main/kotlin/app/revanced/patches/youtube/general/suggestions/patch/SuggestionsShelfPatch.kt#L60

But it doesn't hide suggestions shelf header, so he is using bytebuffer filter to hide header. Should I stick with current implementation or try to find the way to hide header using current LithoFilterPatch.

**Comment by oSumAtrIX** [comment]
Using a hook for the current navigation bar item should be better and useful for the future too

**Comment by johnconner122** [comment]
> Is this ready for review and merge?
> 
> It seems like the outstanding issues are resolved.

Whenever we are in library tab, and open download page, or history page, or any other page in library tab, and then go back, back pressed will be called and index will go back to previous one, but we are still in library tab, amd index shouldn't have changed.

**Comment by n1cKz1** [comment]
"watch again" on 18.38.44 now shows up as horizontal_tile_shelf.eml instead of horizontal_video_shelf.eml. Might be the case for the rest of the components too but they are not showing up for me right now so can't tell

**Comment by oSumAtrIX** [comment]
Hey, just letting you know that we have renamed repositories. Please close this PR and re-open it [here](https://github.com/ReVanced/revanced-patches/pulls). PS: No additional changes are needed to your remotes.

**Comment by LisoUseInAIKyrios** [comment]
> Whenever we are in library tab, and open download page, or history page, or any other page in library tab, and then go back, back pressed will be called and index will go back to previous one, but we are still in library tab, and index shouldn't have changed.

@johnconner122
This should now be fixable using `NavigationBarHookPatch` which shows which tab is active.  Want to try incorporating that with these changes?

**Comment by johnconner122** [comment]
> @johnconner122 This should now be fixable using `NavigationBarHookPatch` which shows which tab is active. Want to try incorporating that with these changes?

I have already incorporated these changes. Seems to be working fine, I'll open a PR later.

**Pull request closed without merging**